arangod/RocksDBEngine/Methods/RocksDBTrxMethods.cpp

```diff
@@ -99,12 +99,15 @@ void RocksDBTrxMethods::rollbackOperation(
   }
 }
 
-Result RocksDBTrxMethods::checkIntermediateCommit(
-    bool& hasPerformedIntermediateCommit) {
-  // perform an intermediate commit if necessary
-  size_t currentSize =
-      _rocksTransaction->GetWriteBatch()->GetWriteBatch()->GetDataSize();
-  return checkIntermediateCommit(currentSize, hasPerformedIntermediateCommit);
+Result RocksDBTrxMethods::checkIntermediateCommit() {
+  Result res;
+  if (hasIntermediateCommitsEnabled()) {
+    // perform an intermediate commit if necessary
+    size_t currentSize =
+        _rocksTransaction->GetWriteBatch()->GetWriteBatch()->GetDataSize();
```

Comment:
do you get a write batch from a write batch?

Reply:
* `_rocksTransaction` is a `rocksdb::Transaction*`.
* `rocksdb::Transaction::GetWriteBatch()` returns the transaction's `rocksdb::WriteBatchWithIndex*`
* `rocksdb::WriteBatchWithIndex::GetWriteBatch()` return's the WBWI's internal `rocksdb::WriteBatch*`.

Reply:
ah ok it was confusing because of their naming